dss/operations/ssm_params.py

```diff
@@ -0,0 +1,161 @@
+"""
+Get and set parameters in the SSM parameter store "environment". Parameters are stored as key-value
+pairs in a map named "environment" (stored in the SSM param store).
+"""
+import os
+import sys
+import select
+import json
+import argparse
+import logging
+import typing
+
+from botocore.exceptions import ClientError
+
+from dss.operations import dispatch
+import dss.operations.util as util
+from dss.util.aws.clients import ssm as ssm_client  # type: ignore
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_ssm_variable_prefix() -> str:
+    """Use info from local environment to assemble necessary prefix for SSM param store variables"""
+    store_name = os.environ["DSS_PARAMETER_STORE"]
+    stage_name = os.environ["DSS_DEPLOYMENT_STAGE"]
+    store_prefix = f"{store_name}/{stage_name}"
+    return store_prefix
+
+
+def fix_ssm_variable_prefix(param_name: str) -> str:
+    """Add the variable store and stage prefix to the front of an SSM param name"""
+    prefix = get_ssm_variable_prefix()
+    if not (param_name.startswith(prefix) or param_name.startswith("/" + prefix)):
+        param_name = f"{prefix}/{param_name}"
+    return param_name
+
+
+def get_ssm_environment() -> dict:
+    """Get the value of the parameter named "environment" in the SSM param store"""
+    prefix = get_ssm_variable_prefix()
+    p = ssm_client.get_parameter(Name=f"/{prefix}/environment")
+    parms = p["Parameter"]["Value"]  # this is a string, so convert to dict
+    return json.loads(parms)
+
+
+def set_ssm_environment(env: dict) -> None:
+    """
+    Set the SSM param store param "environment" to the values in env (dict).
+
+    :param env: dict containing environment variables to set in SSM param store
+    :returns: nothing
+    """
+    prefix = get_ssm_variable_prefix()
+    ssm_client.put_parameter(
+        Name=f"/{prefix}/environment", Value=json.dumps(env), Type="String", Overwrite=True
+    )
+
+
+def set_ssm_parameter(env_var: str, value) -> None:
+    """
+    Set a parameter in the SSM param store variable "environment".
+
+    :param env_var: the name of the environment variable being set
+    :param value: the value of the environment variable being set
+    """
+    environment = get_ssm_environment()
+    try:
```

Comment:
Is there a reason this is not `prev_value = environment.get(env_var)`?

Reply:
No, just some inefficient code. Thank you for the suggestion!